### DISCUSSION TITLE
Where to get xmake.lsp

### DISCUSSION THREAD

**Discussion by Ca1len** [Mythos-404/xmake.nvim#17] [Q&A] [answered]
I apologize for the non-template issue. I was unable to locate the correct suggestion for my question. Could you please advise where I can obtain the xmake.lsp module from tags/3.0.7 xmake.nvim/lua/xmake/init.lua line 22? I have been unable to find any information on this matter.

Plugin was added to my configuration via Lazy, utilizing the following code in a separate file:
```lua
return {
  {
    "Mythos-404/xmake.nvim",
    version = "3.0.7",
    lazy = true,
    event = "BufReadPost",
    config = true,
  },
}
```

When I opened xmake.lua from my project, I encountered the following error:
```
05:00:48 PM msg_show Error executing lua callback: ...hin/.local/share/nvim/lazy/xmake.nvim/lua/xmake/init.lua:6: module 'xmake.lsp' not found:
	no field package.preload['xmake.lsp']
cache_loader: module xmake.lsp not found
cache_loader_lib: module xmake.lsp not found
	no file './xmake/lsp.lua'
	no file '/usr/share/luajit-2.1/xmake/lsp.lua'
	no file '/usr/local/share/lua/5.1/xmake/lsp.lua'
	no file '/usr/local/share/lua/5.1/xmake/lsp/init.lua'
	no file '/usr/share/lua/5.1/xmake/lsp.lua'
	no file '/usr/share/lua/5.1/xmake/lsp/init.lua'
	no file './xmake/lsp.so'
	no file '/usr/local/lib/lua/5.1/xmake/lsp.so'
	no file '/usr/lib64/lua/5.1/xmake/lsp.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file './xmake.so'
	no file '/usr/local/lib/lua/5.1/xmake.so'
	no file '/usr/lib64/lua/5.1/xmake.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
	[C]: in function 'require'
	...hin/.local/share/nvim/lazy/xmake.nvim/lua/xmake/init.lua:6: in function '__index'
	...rshin/.local/share/nvim/lazy/xmake.nvim/plugin/xmake.lua:28: in function <...rshin/.local/share/nvim/lazy/xmake.nvim/plugin/xmake.lua:26>
	[C]: in function 'nvim_exec_autocmds'
	/usr/share/nvim/runtime/lua/vim/lsp/client.lua:948: in function '_on_attach'
	/usr/share/nvim/runtime/lua/vim/lsp/client.lua:616: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by Mythos-404** [comment]
This function is still under testing. I accidentally submitted the test code 和 it has been fixed :)

**Comment by Ca1len** [comment]
Thank you for the answer! May be you can help me with my off-topic main question: is it possible to enable LSP for the xmake.lua in local project?

**Reply by Mythos-404** [reply]
Is that the cpp or lua LSP?

**Reply by Ca1len** [reply]
It's LUA lsp for xmake.lsp file. CPP LSP works well

**Reply by Mythos-404** [reply, marked as answer]
`lua-lsp` You can take a look at the `xmake addons` in [`LLS-Addons`](https://github.com/LuaLS/LLS-Addons) but it's pretty old, right now I'm re`xmake addons` and auto-injecting them into `lua-lsp`!

**Reply by Ca1len** [reply]
Thank you very much for your support! I'll give it a try and hope it helps! I think this discussion can be closed if no other solution can be proposed)
